packages/plugin-http-client/src/rest/AbstractResource.ts

```diff
@@ -0,0 +1,170 @@
+import { GenericError, Dependencies } from '@ima/core';
+
+declare module '@ima/core' {
+  interface OCAliasMap {
+    REST_CLIENT_BASE_API_URL: string;
+  }
+}
+
+import {
+  HttpClient,
+  HttpClientRequestMethod,
+  HttpClientRequestOptions,
+} from '../HttpClient';
+
+export class AbstractResource {
+  #httpClient: HttpClient;
+  #baseApiUrl: string;
+
+  static get $dependencies(): Dependencies {
+    return [HttpClient, '?REST_CLIENT_BASE_API_URL'];
+  }
+
+  static get PathType() {
```

Comment:
nemusi byt getter, muze byt staticka promenna, je to kratsi

Reply:
Když by byla staticka proměnná, tak se dá přepsat v JS, takto jako getter to v JS nepřepíšeš - tím, že je to seznam konstant, tak je toasi lepší nepřepsatelné.